### DISCUSSION TITLE
add EVAL_RO & EVALSHA_RO

### DISCUSSION THREAD

**Discussion by xiaobiaozhao** [apache/kvrocks#720] [General]
1. Each worker thread contains a Lua VM, the Lua VM shared with all connections on this thread. This VM specifically executes the EVAL_RO command.
2. Each connection's commends is pipeline runing.
3. Each worker's connetions send a lot of commands to kvrocks at the same time, but the same thread will exec command one by one. So the same thread can share one Lua VM.

![image](https://user-images.githubusercontent.com/52393536/177780481-05ae59e9-8778-4f69-8f6a-0f5387e75b0f.png)

**Comment by git-hulk** [comment]
Cool, it's good to separate the Lua vm to improve the performance in the readonly mode.
You can go ahead implementing this if you're willing to do this.
